src/main/java/com/hm/achievement/db/AbstractSQLDatabaseManager.java

```diff
@@ -542,4 +530,40 @@ public void clearConnection(UUID uuid) {
 	protected String getPrefix() {
 		return prefix;
 	}
-}
+
+	/**
+	 * Returns a clean list of Achievements get by a player.
+	 *
+	 * @param uuid UUID of a player.
+	 * @return ArrayList containing all information about achievements awarded to a player.
+	 */
+	public List<AwardedDBAchievement> getCleanPlayerAchievementsList(UUID uuid) {
```

Comment:
Could this new method be used in _BookCommand_? Also, wouldn't it be a good idea to use it in the API as well, and deprecate the existing `getPlayerAchievementsList` and _Achievement_ class?

Reply:
> would require some call inside AwardedDBAchievement that returns a formatted Date string

Currently the date is formatted with a private formatter in AbstractSQLDatabaseManager and I don't know how you would like to handle that. 
The new formatting should probably be accessible via a static method so that a reference to it isn't required in every achievement object (alternatively static final class variable) 

I think it would make more sense to make the new class extend the API class so that it can be returned directly to the API call (& doesn't break the API), but currently the new class can not return a formatted date.

The old messy list in database should indeed be deprecated after the formatting is possible.

Reply:
What if _AwardedDBAchievement_ held a formatted date string rather than a `Date` object? That would probably simplify the problem quite a bit.

Reply:
That would also limit what can be done with the date.  
Now that you mention it it's probably dumb to store Date object when a epoch ms would do. I'll try to manage something for it.

Reply:
As an alternative, you could store both an epoch and an "Advanced Achievement's officially formatted string". ^^

Reply:
Stored both long & formatted string in *AwardedDBAchievement* also added a toAPIAchievement method because ...API.Achievement was a final class & could not be extended.

Also Deprecated `getPlayerAchievementsList` after replacing the calls to it with the new implementation

Reply:
I would be inclined to remove the `getPlayerAchievementsList` completely rather than deprecating it. It is not part of the API and an API alternative exists, so most likely nobody is using it. It's just dead code at the moment.

You could subsequently rename `getCleanPlayerAchievementsList` to the old method's name. I find it somewhat confusing for a method to claim it is "clean". 😃